packages/x-charts-pro/src/Heatmap/Heatmap.tsx

```diff
@@ -0,0 +1,187 @@
+import * as React from 'react';
+import useId from '@mui/utils/useId';
+import { ChartsAxis, ChartsAxisProps } from '@mui/x-charts/ChartsAxis';
+import {
+  ChartsTooltip,
+  ChartsTooltipProps,
+  ChartsTooltipSlotProps,
+  ChartsTooltipSlots,
+} from '@mui/x-charts/ChartsTooltip';
+import {
+  MakeOptional,
+  ChartsAxisSlots,
+  ChartsAxisSlotProps,
+  ChartsXAxisProps,
+  ChartsYAxisProps,
+  AxisConfig,
+} from '@mui/x-charts/internals';
+import { ChartsClipPath } from '@mui/x-charts/ChartsClipPath';
+import {
+  ChartsOnAxisClickHandler,
+  ChartsOnAxisClickHandlerProps,
+} from '@mui/x-charts/ChartsOnAxisClickHandler';
+import {
+  ChartsOverlay,
+  ChartsOverlayProps,
+  ChartsOverlaySlotProps,
+  ChartsOverlaySlots,
+} from '@mui/x-charts/ChartsOverlay';
+import {
+  ResponsiveChartContainerPro,
+  ResponsiveChartContainerProProps,
+} from '../ResponsiveChartContainerPro';
+import { HeatmapSeriesType } from '../models/seriesType/heatmap';
+import { HeatmapPlot } from './HeatmapPlot';
+import { plugin as heatmapPlugin } from './plugin';
+import { DefaultHeatmapTooltip } from './DefaultHeatmapTooltip';
+
+export interface HeatmapSlots
+  extends ChartsAxisSlots,
+    Omit<ChartsTooltipSlots<'heatmap'>, 'axisContent'>,
+    ChartsOverlaySlots {}
+export interface HeatmapSlotProps
+  extends ChartsAxisSlotProps,
+    Omit<ChartsTooltipSlotProps<'heatmap'>, 'axisContent'>,
+    ChartsOverlaySlotProps {}
+
+export interface HeatmapProps
+  extends Omit<ResponsiveChartContainerProProps, 'series' | 'plugins' | 'xAxis' | 'yAxis'>,
+    Omit<ChartsAxisProps, 'slots' | 'slotProps'>,
+    Omit<ChartsOverlayProps, 'slots' | 'slotProps'>,
+    ChartsOnAxisClickHandlerProps {
+  /**
+   * The configuration of the x-axes.
+   * If not provided, a default axis config is used.
+   * An array of [[AxisConfig]] objects.
+   */
+  xAxis: MakeOptional<AxisConfig<'band', any, ChartsXAxisProps>, 'id' | 'scaleType'>[];
+  /**
+   * The configuration of the y-axes.
+   * If not provided, a default axis config is used.
+   * An array of [[AxisConfig]] objects.
+   */
+  yAxis: MakeOptional<AxisConfig<'band', any, ChartsYAxisProps>, 'id' | 'scaleType'>[];
+  /**
+   * The series to display in the bar chart.
+   * An array of [[HeatmapSeriesType]] objects.
+   */
+  series: MakeOptional<HeatmapSeriesType, 'type'>[];
+  /**
+   * The configuration of the tooltip.
+   * @see See {@link https://mui.com/x/react-charts/tooltip/ tooltip docs} for more details.
+   */
+  tooltip?: ChartsTooltipProps<'heatmap'>;
+  /**
+   * Overridable component slots.
+   * @default {}
+   */
+  slots?: HeatmapSlots;
+  /**
+   * The props used for each component slot.
+   * @default {}
+   */
+  slotProps?: HeatmapSlotProps;
+}
+
+export const Heatmap = React.forwardRef(function Heatmap(props: HeatmapProps, ref) {
```

Comment:
Do you think it would be reasonable to remove the "black" lines between each cell?

![Screenshot 2024-06-17 at 17 14 17](https://github.com/mui/mui-x/assets/9891233/31a0cc4a-906d-4746-9bbe-5656c875975c)

Reply:
Maybe by using `shape-rendering: crispedges;`